<launch>

    <!-- gesture_recognition.launch -->
    <!-- Launches the hand sign recognition node for gesture recognition. -->
    
    <!-- Arguments -->
    <arg name="ns" default="hand_steer"/>

    <arg name="keypoint_classifier_label"
        default="$(find hand_steer_sim)/hand_steer_sim/model/keypoint_classifier/keypoint_classifier_label.csv"/>

    <arg name="keypoint_classifier_model"
        default="$(find hand_steer_sim)/hand_steer_sim/model/keypoint_classifier/keypoint_classifier.tflite"/>

    <arg name="subscribe_image_topic" default="/image_raw"/>
    <arg name="publish_gesture_topic" default="/gesture/hand_sign"/>
    <arg name="show_image" default="True"/>

    <group ns="$(arg ns)">
        <node name="hand_sign_recognition" pkg="hand_steer_sim" type="hand_sign_recognition_node.py" output="screen">
            <param name="keypoint_classifier_label" type="string" value="$(arg keypoint_classifier_label)" />
            <param name="keypoint_classifier_model" type="string" value="$(arg keypoint_classifier_model)" />
            <param name="subscribe_image_topic" type="string" value="$(arg subscribe_image_topic)" />
            <param name="publish_gesture_topic" type="string" value="$(arg publish_gesture_topic)" />
            <param name="show_image" type="bool" value="$(arg show_image)" />
        </node>
    </group>

</launch>